models/sandpile/sandpile.cpp

```diff
@@ -14,16 +14,24 @@ WARPED_REGISTER_POLYMORPHIC_SERIALIZABLE_CLASS(SandEvent)
 
 std::string node_name (unsigned int index) {
 
-    return std::string("Node_") + std::to_string(index);
+    return std::string("Vertex_") + std::to_string(index);
 }
 
 std::vector<std::shared_ptr<warped::Event> > Vertex::initializeLP() {
 
     std::vector<std::shared_ptr<warped::Event>> events;
 
-    /* If z-value exceeds threshold, send an event to itself and 4 neighbors */
-    if (state_.z_ >= stability_threshold_) {
-        events.emplace_back( new SandEvent {node_name(this->index_)                 , TS_INTERVAL} );
+    /* If there is no grain of sand */
+    if (!state_.sandpile_height_) return events;
+
+    /* If the pile is still stable */
+    if (state_.sandpile_height_ < 3) {
```

Comment:
Should this check if the height is greater than the threshold rather than a constant?